packages/protocol/lib/compatibility/verify-bytecode.ts

```diff
@@ -78,60 +81,63 @@ const getOnchainBytecode = async (address: string, context: VerificationContext)
 const isLibrary = (contract: string, context: VerificationContext) =>
   contract in context.libraryAddresses.addresses
 
-const getImplementationAddress = async (contract: string, context: VerificationContext) => {
-  // Where we find the implementation address depends on two factors:
-  // 1. Is the contract affected by a governance proposal?
-  // 2. Is the contract registered in the Registry or a linked library?
-
-  if (isImplementationChanged(contract, context)) {
-    return getProposedImplementationAddress(contract, context)
-  }
-
-  if (isLibrary(contract, context)) {
-    return `0x${context.libraryAddresses.addresses[contract]}`
-  }
-
-  // contract is registered but we need to check if the proxy is affected by the proposal
-  const proxyAddress = isProxyChanged(contract, context)
-    ? getProposedProxyAddress(contract, context)
-    : await context.registry.getAddressForString(contract)
-
-  // at() returns a promise despite Typescript labelling the await as extraneous
-  const proxy: ProxyInstance = await context.Proxy.at(
-    context.web3.utils.toChecksumAddress(proxyAddress)
-  )
-  return proxy._getImplementation()
-}
-
 const dfsStep = async (queue: string[], visited: Set<string>, context: VerificationContext) => {
   const contract = queue.pop()
   // mark current DFS node as visited
   visited.add(contract)
 
-  // get source code
+  // check proxy deployment
+  if (isProxyChanged(contract, context)) {
+    const proxyAddress = getProposedProxyAddress(contract, context)
+    const governanceAddr = await context.registry.getAddressForString('Governance')
```

Comment:
Optional, slight optimalization: we could get the Governance address once and store it in context.

Reply:
now that I think of it, the governance address could change but I will make this optimization for now